src/test/scala/sigmastate/serialization/DeserializationResilience.scala

```diff
@@ -47,4 +64,135 @@ class DeserializationResilience extends SerializationSpecification {
     an[InvalidOpCode] should be thrownBy
       ValueSerializer.deserialize(Array.fill[Byte](1)(117.toByte))
   }
+
+  private def traceReaderCallDepth(expr: SValue): (IndexedSeq[Int], IndexedSeq[Int]) = {
+    class LoggingSigmaByteReader(r: Reader) extends
+      SigmaByteReader(r,
+        new ConstantStore(),
+        resolvePlaceholdersToConstants = false,
+        maxTreeDepth = SigmaSerializer.MaxTreeDepth) {
+      val levels: mutable.ArrayBuilder[Int] = mutable.ArrayBuilder.make[Int]()
+      override def level_=(v: Int): Unit = {
+        if (v >= super.level) {
+          // going deeper (save new depth to account added depth by the caller)
+          levels += v
+        } else {
+          // going up (save previous depth to account added depth be the caller)
```

Comment:
This comment is not clear

Reply:
Thanks! I rephrased the comment. 
https://github.com/ScorexFoundation/sigmastate-interpreter/blob/3818e6f90c56a12080ac2de17078e91fc42a75f8/src/test/scala/sigmastate/serialization/DeserializationResilience.scala#L80